ci-operator/step-registry/cucushift/upgrade/check/capability/cucushift-upgrade-check-capability-commands.sh

```diff
@@ -0,0 +1,451 @@
+    "None")
+    ;;
+    "v4.11")
+    caps_string="baremetal marketplace openshift-samples"   
+    ;;
+    "v4.12")
+    caps_string="baremetal marketplace openshift-samples Console Insights Storage CSISnapshot"
+    ;;
+    "v4.13")
+    caps_string="baremetal marketplace openshift-samples Console Insights Storage CSISnapshot NodeTuning"
+    ;;
+    "v4.14")
+    caps_string="baremetal marketplace openshift-samples Console Insights Storage CSISnapshot NodeTuning MachineAPI Build DeploymentConfig ImageRegistry"
+    ;;
+    "v4.15")
+    caps_string="baremetal marketplace openshift-samples Console Insights Storage CSISnapshot NodeTuning MachineAPI Build DeploymentConfig ImageRegistry OperatorLifecycleManager"
+    ;;
+    *)
+    caps_string="baremetal marketplace openshift-samples Console Insights Storage CSISnapshot NodeTuning MachineAPI Build DeploymentConfig ImageRegistry OperatorLifecycleManager"
+    ;;
+    esac
+
+    if (( minor_version >=14 && major_version == 4 )) && [[ $caps_string != *"MachineAPI"* ]]; then
+        caps_string="${caps_string} MachineAPI"
+    fi
+    echo $caps_string
+}
+
+# Extract manifests from release
+function extract_manifests() {
+    # echo "Extracing manifests..."
+    local work_dir="$1"
+    local release="$2"
+    #local tmp_known_caps
+
+    # Check if the directory exists
+    if [ ! -d "${work_dir}" ] || [ -z "${release}" ]; then
+        echo "ERROR: Missing required arguments"
+        return 1
+    fi
+
+    # Extract manifests
+    if ! oc adm release extract -a "${CLUSTER_PROFILE_DIR}/pull-secret" --to ${work_dir} --from ${release}; then
+        echo "ERROR: Extracting manifest failed"
+        return 1
+    fi
+}
+
+# Get implicitly enabled capabilities from cluster
+function get_actual_implicit_caps {
+    # echo "Getting actual implicitly enabled caps in cluster..."
+    local tmp_capabilities implicit_status implicit_message
+    # Obtain ImplicitlyEnabledCapabilities condition status from cluster
+    if ! implicit_status=$(oc get clusterversion version -o json | jq -r '.status.conditions[] | select(.type == "ImplicitlyEnabledCapabilities").status'); then
+        echo "ERROR: Get implicitly enabled caps failed"
+        return 1
+    fi
+    if [ "$implicit_status" = "False" ]; then
+        echo "INFO: No implicitly enabled capabilities in the cluster"
+        return
+    fi
+    # Obtain ImplicitlyEnabledCapabilities condition message from cluster
+    if ! implicit_message=$(oc get clusterversion version -o json | jq -r '.status.conditions[] | select(.type == "ImplicitlyEnabledCapabilities").message'); then
+        echo "ERROR: Get implicitly enabled caps failed"
+        return 1
+    fi
+
+    # Parse the message to extract implicit caps
+    # The condition message could be "The following capabilities could not be disabled: Console, Insights, Storage"
+    tmp_capabilities=$(echo "$implicit_message" | grep -oE 'could not be disabled: (.*)' | cut -d':' -f2)
```

Comment:
```suggestion
    tmp_capabilities=$(echo "$implicit_message" | grep -oE 'could not be disabled: (.*)' | cut -d': ' -f2)
```

Reply:
Updated.

Reply:
Reverted back as cut only accepts one character of delimiter.